### PR TITLE
release debug pins on HC32 boards

### PR DESCRIPTION
**Pull request by shadow578**
### Description

On ~some~ all currently supported HC32 boards, pins normally assigned to debugging functions are re-used for other functions.
The PR implements `DISABLE_DEBUG` and `DISABLE_JTAG` in the HC32 HAL and adds the corrosponding defined to the HC32 board files.

Currently, this doesn't cause issues since the bootloader releases these pins before booting into Marlin.
However, it's probably best to not rely on a side-effect of the bootloader, in case it changes with newer board revisions (or the user changes it).

### Requirements

HC32-Based Board

### Benefits

Removes reliance on side-effect of the bootloader

### Configurations

N/A

### Related Issues

N/A

### Appendix

`PSPCR` register overview (set by `PORT_DebugPortSetting`)

| grafik |
|-|
| Except from HC32F460 Reference Manual, Section 9.4.7 Special Control Register (PSPCR) |

Debug Port Pins assignments according to HC32F460 Datasheet

Pin | Debug Function
-|-
PA13 | JTMS / SWDIO
PA14 | JTCK / SWCLK
PA15 | JTDI
PB3 | JTDO
PB4 | NJTRST

### CODE REVIEW

File: Marlin/src/HAL/HC32/HAL.h [hunk 1]
```diff
@@ -148,6 +148,18 @@
...
+// Debug port disable
+// JTMS / SWDIO = PA13
+// JTCK / SWCLK = PA14
+// JTDI         = PA15
+// JTDO         = PB3
+// NJTRST       = PB4
+//
+#define JTAGSWD_RESET() PORT_DebugPortSetting(0x1F, Enable);    // enable JTDO, JTDI, NJTRST, SWDIO, SWCLK
+#define JTAG_DISABLE() PORT_DebugPortSetting(0x1C, Disable);    // disable JTDO, JTDI, NJTRST
+#define JTAGSWD_DISABLE() PORT_DebugPortSetting(0x1F, Disable); // disable JTDO, JTDI, NJTRST, SWDIO, SWCLK
```
Comment (sjasonsmith) [inline Marlin/src/HAL/HC32/HAL.h]: What do you think of using the named variables here for the pins?

```suggestion
#define JTAGSWD_RESET() PORT_DebugPortSetting(ALL_DBG_PIN, Enable);
#define JTAG_DISABLE() PORT_DebugPortSetting(TDO_SWO | TDI | TRST, Disable);
#define JTAGSWD_DISABLE() PORT_DebugPortSetting(ALL_DBG_PIN, Disable);
```

Comment (shadow578) [reply Marlin/src/HAL/HC32/HAL.h]: looks a lot more readable than using "magic" values. I'll update the PR accordingly

Comment (sjasonsmith) [reply Marlin/src/HAL/HC32/HAL.h]: Sorry for the delay, I had seen your comment but didn't realize the change has been made.

Comment (sjasonsmith) [reply Marlin/src/HAL/HC32/HAL.h]: The bit names I used in my recommendation were actually from the HC32 headers, I don't think you had to redefine them.

That said, your bit names are a little more consistent than those, so I think it is fine to go ahead and use what you've done in your most recent change.

**Pull request merged**